### PR TITLE
refactor: remove codeclient dependancy [IDE-364]

### PR DESCRIPTION
**Pull request by ShawkyZ**
### Description
- Removed snyk/code-client dependency

### Checklist

- [ ] Tests added and all succeed
- [x] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/snyk/snykCode/utils/ip.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import * as dns from 'dns';
+import { promisify } from 'util';
+
+const lookupAsync = promisify(dns.lookup);
```
Comment (teodora-sandu) [inline src/snyk/snykCode/utils/ip.ts:4]: Question: Why did you choose to make this async? I guess it was an async function before but would it not have been better to just change the code that calls it to stop using `await`?

File: src/test/unit/base/services/authenticationService.test.ts [hunk 2]
```diff
@@ -71,48 +71,22 @@ suite('AuthenticationService', () => {
...
-      // eslint-disable-next-line camelcase
-      return needle.post(uri, data, { ...opts, ...overrideNeedleTimeoutOptions });
     });
 
     const ipFamily = await getIpFamily('https://dev.snyk.io');
 
     strictEqual(ipFamily, undefined);
   });
 
-  test('getIpFamily returns 6 when IPv6 supported', async () => {
```
Comment (teodora-sandu) [inline src/test/unit/base/services/authenticationService.test.ts:94]: I think you could have kept this test and changed it to be for your new `getIpFamily` function to show backwards compatibility

File: src/snyk/snykCode/constants/dcignore.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+export declare const CACHE_KEY = '.dccache';
+export declare const MAX_UPLOAD_ATTEMPTS = 10;
+export declare const UPLOAD_CONCURRENCY = 2;
+export declare const POLLING_INTERVAL = 500;
+export declare const MAX_RETRY_ATTEMPTS = 10;
+export declare const REQUEST_RETRY_DELAY: number;
+export declare const ORG_ID_REGEXP: RegExp;
+export declare const IGNORES_DEFAULT: string[];
+export declare const IGNORE_FILES_NAMES: string[];
+export declare const DCIGNORE_DRAFTS: {
```
Comment (teodora-sandu) [inline src/snyk/snykCode/constants/dcignore.ts:19]: I think you may have copied these over and formatted them incorrectly from https://github.com/snyk/code-client/blob/master/src/constants.ts. I was following https://docs.snyk.io/snyk-scm-ide-and-ci-cd-integrations/use-snyk-in-your-ide/visual-studio-code-extension/create-a-.dcignore-file to try and use this command that gets run with the constants and I'm seeing an error
Screenshot 2024-06-10 at 12 46 31

This file here defined the types of these constants but does not assign a value to them. When I copy that file directly over to this file it works so it should be an easy fix

**Review by teodora-sandu — COMMENTED** [review]
The command doesn't work anymore so I've left a comment with what I think the issue is. Could you please add a CHANGELOG and also change the commit message to a `refactor`, since `chore`s are not meant to change any of the code running for customers.

File: src/test/unit/base/services/authenticationService.test.ts (hunk 2 above)
Comment (ShawkyZ) [reply src/test/unit/base/services/authenticationService.test.ts:94]: I'm wondering if we still need the other test though. wdyt?

Comment (teodora-sandu) [reply src/test/unit/base/services/authenticationService.test.ts:94]: This one https://github.com/snyk/vscode-extension/blob/f3b8e34d5e26e4ae8c3eabe74b11399ba0ce8274/src/test/unit/base/services/authenticationService.test.ts#L74? I agree with you that before we shouldn't have really tested this function here, it should have been in `snyk/code-client`. But now that you implemented that function in the extension then it makes sense to keep both tests

**Review by teodora-sandu — APPROVED** [review]

**Pull request merged**